PurchasesTests/Networking/BackendTests.swift

```diff
@@ -718,6 +731,46 @@ class BackendTests: XCTestCase {
         expect(eligibility!["productc"]!.status).toEventually(equal(IntroEligibilityStatus.unknown))
     }
 
+    func testEligibilityUnknownIfMissingAppUserID() {
+        // Set us up for a 404 because if the input sanitizing code fails, it will execute and we'd get a 404.
+        let response = HTTPResponse(statusCode: 404, response: nil, error: nil)
+        let path = "/subscribers//intro_eligibility"
+        httpClient.mock(requestPath: path, response: response)
+
+        var eligibility: [String: IntroEligibility]?
+        let products = ["producta"]
+        var eventualError: NSError?
+        backend?.getIntroEligibility(appUserID: "", receiptData: Data.init(1...2), productIdentifiers: products, completion: {(productEligibility, error) in
+            eventualError = error as NSError?
+            eligibility = productEligibility
+        })
+
+        expect(eligibility!["producta"]!.status).toEventually(equal(IntroEligibilityStatus.unknown))
+        expect(eventualError).toEventuallyNot(beNil())
+        expect(eventualError?.domain).to(equal(RCPurchasesErrorCodeDomain))
+        expect(eventualError?.localizedDescription).to(equal(ErrorUtils.missingAppUserIDError().localizedDescription))
+
+        // We only have an underlyingError if we got a response from the server. This shouldn't happen because
+        // we shouldn't even send a request.
+        var underlyingError = (eventualError?.userInfo[NSUnderlyingErrorKey]) as? NSError
+        expect(underlyingError).to(beNil())
+
+        backend?.getIntroEligibility(appUserID: "   ", receiptData: Data.init(1...2), productIdentifiers: products, completion: {(productEligibility, error) in
```

Comment:
let's split this one up into one line per param

Reply:
There were a few like this I took care of.